### DISCUSSION TITLE
How to disable pylance checking in vscode as ruff is already taking care of it?

### DISCUSSION THREAD

**Discussion by whoisarpit** [astral-sh/ruff#4128] [Q&A] [answered]
Apologies for posting here, [ruff-vscode](https://github.com/charliermarsh/ruff-vscode) doesn't have a QnA page.

Since I'm running my vscode on a shared remote server, I'd like to avoid as much extra computation as possible and pylance takes a lot of compute resources.

![image](https://user-images.githubusercontent.com/17565234/234751831-a7141fb4-9a13-4c4f-ab03-26b0da4190fc.png)

**Comment by whoisarpit** [comment, marked as answer]
Found a solution for reference:

In `pyproject.toml`, add
```toml
[tool.pyright]
typeCheckingMode = "off"
```

**Reply by charliermarsh** [reply]
Here's another solution that I saw on `ruff-vscode`: https://github.com/charliermarsh/ruff-vscode/issues/133#issuecomment-1455085053

**Reply by charliermarsh** [reply]
(You probably don't want to enable Pylance _entirely_, since Ruff isn't a generic language server -- so it can't do jump-to-completion and other things. So it's useful to have both enabled, but you're right that it can be helpful to avoid duplicate diagnostics.)
